### ISSUE TITLE
Metadata Server Issues (429) & Authors with missing books / Authors not found

### ISSUE THREAD

**Issue by AP0282** [Readarr/Readarr#2783]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

I type in a writers name (Stephen King, Jane Austen ect), they show up, I add them, no books found so not searching for that writer.

### Expected Behavior

I would like it to find the books of those authors!

### Steps To Reproduce

Click "Add New"
Type artist "Stephen King"
Click "Add"
Go to athor in list
No books there at all

### Environment

```markdown
- OS: Windows 10 Pro
- Readarr:0.3.1.2096
- Docker Install: ?
- Using Reverse Proxy: ?
- Browser: Chrome
- Database: Sqlite 3.36.0
```

### What branch are you running?

Master

### Trace Logs?

[Readarr.1.txt](https://github.com/Readarr/Readarr/files/12315400/Readarr.1.txt)

### Trace Logs have been provided as applicable. Reports may be closed if the required logs are not provided.

- [X] I have read and followed the steps in the wiki link above and provided the required trace logs - the logs contain `trace` - that are relevant and show this issue.

**Comment by AP0282** [comment]
I apologise if above is not 100% correct as I am very much a newbie when it comes to this type of stuff.

**Comment by mynameisbogdan** [comment]
Currently authors with a lot of books and editions cannot be added to Readarr. This is a temporary situation that the devs are aware of. There is nothing you can do at the moment to add these authors or books.

Feel free to join us on Discord for more details.

**Comment by bakerboy448** [comment]
For posterity this has also spread to generally be impacting the metadata server as a whole with intermittent ongoing issues and general instability

See https://github.com/Readarr/Readarr/issues/2970#issuecomment-1822031496

**Comment by egyptianbman** [comment]
I'm not sure if this is a contributor to instability, but I noticed that the BookInfoProxy continues to report `BookInfo returned 429, backing off for 10s` even after the frontend reports failure and stops checking for a response. With the backend continuously querying for book info, if the user refreshes multiple times, the back end may be sending excessive requests that eventually won't be consumed by the frontend. I also noticed that the same happens if the user navigates away from the page.

**Comment by bakerboy448** [comment]
https://wiki.servarr.com/readarr/system#we-are-currently-experiencing-issues-with-our-metadata-provider-this-will-prevent-user-searches-and-author-refreshes-from-working-we-are-working-to-solve-the-issue

**Comment by francispeixoto** [comment]
Is this still a thing? Because as of today, i still have incomplete authors and i can't seem to search by name, only by goodreads id and it doesn't always work.

**Comment by Jebula999** [comment]
Hi There,

We now in July and this still appears to be a thing.
Same as @AP0282 I have added Stephen King, but 0 books are returned.

I tried +-5 different Metadata Profiles with no luck

**Comment by bakerboy448** [comment]
Pinned post in discord still applies.

These authors likely will not be fixed anytime in the near future as it will likely require changing metadata providers which is a monumental task

> The current state of the Readarr metadata server: 1) It's functioning, approximately 3 months behind. If you have a book that was added to Goodreads less than 3 months ago, it probably is not going to show up yet. 2) Large authors are still not working, and will not be for a while. A partial, but not exhaustive, list of those authors: Stephen King, Isaac Asimov, Leo Tolstoy, Fyodor Dostoevsky, Arthur Conan Doyle, Stephen Fry, Agatha Christie, Arthur C. Clarke, Oscar Wilde, Jack London, Neil Gaiman, Charles Dickens, Jane Smiley, Vladimir Nabokov, Bernard Cornwall. 3) For large authors, you cannot do anything to add either the author OR their individual books. There is no workaround.

https://discord.com/channels/264387956343570434/1028840624864120872/1230155166334849025

**Comment by Julianwan** [comment]
You guys haven't found a workaround for this? Even as simple as skipping collections or editions on popular authors?

I got readarr / calibre set up today after enjoying the ease of radarr / plex...

And I can't even get a list of Ray Bradbury books... This seems like the entire thing is broken.

**Comment by bakerboy448** [comment]
> You guys haven't found a workaround for this? Even as simple as skipping collections or editions on popular authors?
> 
> I got readarr / calibre set up today after enjoying the ease of radarr / plex...
> 
> And I can't even get a list of Ray Bradbury books... This seems like the entire thing is broken.

...seems you didn't read the current update?

Pinned post in discord still applies.

These authors likely **will not be fixed anytime in the near future as it will likely require changing metadata providers which is a monumental task**

**Comment by bakerboy448** [comment]
> The metadata server is struggling. Something occurred on 7/31 that caused it, and it hasn't been able to be corrected yet. It works. You will see 429's a LOT. Sometimes it's better than other times. Hopefully it will be corrected soon. When that happens, I will update here. There is nothing you can do to help but wait and be patient. It is not an issue on your end.

https://discord.com/channels/264387956343570434/1028840624864120872/1271481982018125855

**Comment by ShyRee1890** [comment]
Noob here. Same issue...searching Stephen King offers me no books.

I'll keep an eye on this thread and hopefully there can be a fix. I really love this app, just wish it worked! :-)

Thanks for all the efforts!

**Comment by HikariNoKitsune** [comment]
Looking around abit, it appears this has been a serious issue for going on a YEAR NOW... Just tossing this out there, but why on earth has no one gone 'heck it' and added something other than the current 'metadata server' into the build? There are so many places to grab metadata from... the most reliable being AUDIBLE... this whole issue seems to be fixable by just adding a different metadata option in there... 

And I know it can be done, cause I'm running Bookshelf for my audiobooks, and it can pull from Google Books, Audible, GoodReads, and a few others.... so I know there is some sorta API out there to just grab metadata info for things... This feels like a super obvious fix for this... please tell me we aren't stuck in some sorta sunken cost fallacy here...?

I'd love to get this sorta thing up and running, as the only other alternative that I can find is LazyLibrarian and I really do not like it with what little i've been toying with... already have most of the other arr services goin, and would love to add this one to the pile... but its kinda just, unusable atm... can't even add in my existing books cause it has to identify them all, which is taking days to do because of this 429 error lol.

Anyway, left my rambly 2 cents I guess.

TLDR: issue is still present, there are obvious fixes for this of which other services for books/audiobooks are using with no issues, and the 'alternatives' to this sort of application/service are pretty crap in comparison to typical arr applications/services. Wish I was better at coding/programming and could help, but I'm pretty useless so all I can saldy really do is complain and gripe behind a keyboard lol.

**Comment by ghost** [comment]
If any of the lurkers here knew how to fix this ourselves, it would have been done already. The problem isn't that it can be done, the problem is developer time.

**Comment by bakerboy448** [comment]
> this whole issue seems to be fixable by just adding a different metadata option in there...

looking forward to your PR to convert readarr and additional efforts to provide a caching central metadata server.....

the long term plan is to likely migrate to OpenLibrary which well is not simple as it requires analysis and migration

further complaints - like yours - are unhelpful, unproductive noise, and add 0 value....they also likely discourage folks from even wanting to help the team with this Herculean task

**Comment by speedyflipper** [comment]
I basically created my own metadata server and changed the dns record for bookinfo to the local server. It has issues but works. I don’t use readarr often so I never put too much effort into once it was functional.

**Comment by J-Swift** [comment]
What dns did you set? Asking for a friend

**Comment by speedyflipper** [comment]
local router. set the api.bookinfo.club to point to whatever local ip

**Comment by n-shay** [comment]
@speedyflipper 
> I basically created my own metadata server

Can you elaborate? Where does your metadata server get the data? Did you emulate all the API endpoints from api.bookinfo.club?  Or do you know if there's a repo for that API somewhere?

**Comment by francispeixoto** [comment]
I'd be down to help if it means I can have a reliable metadata source

**Comment by speedyflipper** [comment]
Yes. I emulated the endpoints and scrap from goodreads

**Comment by bakerboy448** [comment]
https://wiki.servarr.com/en/readarr/metadata-issues

For those not on discord; will try to keep this page current

**Comment by bakerboy448** [comment]
429 Issues Resolved

> Readarr's metadata server appears to be functioning normally with no 429 or cache issues. As always, I will update this pin if things change. All pins are the current state of readarr and if anything changes, the pins will be updated accordingly.

=======

Large Authors issue is covered under 
https://github.com/Readarr/Readarr/issues/3486

**Comment by Mirmentrud** [comment]
Hi, thanks for all your efforts. Unfortunately, I still experience a lot of 429 issues with Reader. Not sure if I missed any configuration, tho..

Version
    0.3.32.2587
Package Version
    0.3.32.2587-ls124 by linuxserver.io 
.NET Core
    Yes (6.0.32)
Docker
    Yes
Database
    Sqlite 3.45.3

**Comment by bakerboy448** [comment]
> The metadata server appears to be having some further issues with 429's. There's not much you can do, although doing ID based searches are better than doing text string searches where possible. The devs have been made aware of the issue (which began on the evening of 9/14), and this pin will updated when there is a change in status. If this pin is up, it's still an issue, no need to ask.

as always see the pinned messages in discord or the wiki page For users coming here from Google. Please see the pinned messages in discord, the pinned GitHub Issues, or the wiki page for current metadata server issues/status

https://wiki.servarr.com/readarr/metadata-issues

**Comment by bakerboy448** [comment]
Please do not share unofficial metadata servers that are not supported by the team.

Work is on going for OpenLibrary as previously noted.
